include/vcpkg/base/system.process.h

```diff
@@ -109,51 +109,98 @@ namespace vcpkg
     Environment get_modified_clean_environment(const std::unordered_map<std::string, std::string>& extra_env,
                                                StringView prepend_to_path = {});
 
-    struct WorkingDirectory
+    struct ProcessLaunchSettings
     {
-        Path working_directory;
+        Command cmd;
+        Optional<Path> working_directory;
+        Optional<Environment> environment;
+
+        ProcessLaunchSettings() = default;
+        ProcessLaunchSettings(const Command& cmd) : cmd(cmd) { }
+        ProcessLaunchSettings(Command&& cmd) : cmd(cmd) { }
+
+        ProcessLaunchSettings& string_arg(StringView s) &
+        {
+            cmd.string_arg(s);
+            return *this;
+        }
+
+        ProcessLaunchSettings& raw_arg(StringView s) &
+        {
+            cmd.raw_arg(s);
+            return *this;
+        }
+
+        ProcessLaunchSettings& forwarded_args(View<std::string> args) &
+        {
+            cmd.forwarded_args(args);
+            return *this;
+        }
+
+        ProcessLaunchSettings&& string_arg(StringView s) && { return std::move(string_arg(s)); };
+        ProcessLaunchSettings&& raw_arg(StringView s) && { return std::move(raw_arg(s)); }
+        ProcessLaunchSettings&& forwarded_args(View<std::string> args) && { return std::move(forwarded_args(args)); }
+
+        StringView command_line() const { return cmd.command_line(); }
     };
 
-    extern const WorkingDirectory default_working_directory;
-    extern const Environment default_environment;
+    struct RedirectedProcessLaunchSettings : ProcessLaunchSettings
```

Comment:
>E.g. cmd_execute(redirected) will do the wrong thing and it isn't terribly obvious.

It isn't clear to me that it does the wrong thing. The only difference between these structs is that the settings in the redirected block are *about* redirection. The only reason they aren't the same struct is so that `cmd_execute` isn't claiming to respect settings it does not.

However, I also don't see any reason for it to ever convert so I changed it by duplicating the members.